core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -513,6 +523,12 @@ public boolean isLegacyDefault() {
          * Plugins in the repository, keyed by their artifact IDs.
          */
         public final Map<String,Plugin> plugins = new TreeMap<String,Plugin>(String.CASE_INSENSITIVE_ORDER);
+        /**
+         * List of warnings (mostly security) published with the update site.
+         *
+         * @since TODO
+         */
+        public final Set<Warning> warnings = new HashSet<Warning>();
```

Comment:
I'd be happy to make them `@Restricted` for now to buy us time to hammer out such details.